src/util.js

```diff
@@ -12,13 +12,58 @@
 //     See the License for the specific language governing permissions and
 // limitations under the License.
 
-(function(internal) {
+(function(internal, testing) {
 
   function inherit(childCtor, parentCtor) {
     childCtor.prototype = Object.create(parentCtor.prototype);
     childCtor.prototype.constructor = childCtor;
   }
 
+  /*
+   * Calls the callback for each value in the object.
+   * @param {!Object} object: The object to iterate over.
+   * @param {Function} callback: A function to be called for each value in the
+   *   object. It should be specified as function(value, key, inputObject).
+   *   Its return value is ignored.
+   * @param {Object} opt_this: The value that will appear as 'this' within the
+   *   callback.
+   */
+  function foreach(object, callback, opt_this) {
```

Comment:
Na, leave it like this. That's a good conventions, I've just never seen it before.